tests/vmi_cloudinit_test.go

```diff
@@ -633,3 +633,14 @@ func lookupCloudInitNoCloudVolume(volumes []v1.Volume) *v1.Volume {
 	}
 	return nil
 }
+
+func withoutCloudInitNoCloud() libvmi.Option {
+	return func(vmi *v1.VirtualMachineInstance) {
+		for i, volume := range vmi.Spec.Volumes {
+			if volume.CloudInitNoCloud != nil {
+				vmi.Spec.Volumes = append(vmi.Spec.Volumes[:i], vmi.Spec.Volumes[i+1:]...)
+				break
```

Comment:
I think we should be more reliable and should remove every cloudinitnocloud volume, without breaking the cycle, even if there could be only one.
In general, I am still non totally convinced about this, but for the moment it could be the better tradeoff.
I'm ready to lgtm once we are sure we are removing all the cloudinitnocloud volumes. (hint: I think we should create a new volume list to be assigned at the end of the for cycle).
If you think that the current implementation is better, I am ready to be convinced :)
Thank you!

Reply:
Thanks for the feedback and reviews, its is much appreciated :slightly_smiling_face: 

There were some discussions offline about having a factory for cirros w/o cloud-init as an alternative, where it create a vmi with no cloudinit columes and the relevant cloudinit volume is added.
Instead of creating a vmi with some cloudinit volume, remove it, and add the relevant cloudinit.

There is a PR that implements it and remove  `NewRandomVMIWithEphemeralDiskAndConfigDriveUserdata` #11923.